### PR TITLE
feat(icon): image src invalid error api

### PR DESCRIPTION
**Pull request by vikalpgupta**
Adding support for error API within the icon component

## Description

Added error API support for icon component having invalid src passed, in such case the component will announce an `Event("error")` which can be handled by parent component

## Related issue(s)

-

## How has this been tested?

-   [x] _Test case 1_
    Added a test case

## Types of changes

-   [ ] Bug fix (non-breaking change which fixes an issue)
-   [x] New feature (non-breaking change which adds functionality)
-   [ ] Breaking change (fix or feature that would cause existing functionality to change)
-   [ ] Chore (minor updates related to the tooling or maintenance of the repository, does not impact compiled assets)

## Checklist

-   [x] I have signed the [Adobe Open Source CLA](http://opensource.adobe.com/cla.html).
-   [x] My code follows the code style of this project.
-   [x] If my change required a change to the documentation, I have updated the documentation in this pull request.
-   [x] I have read the **[CONTRIBUTING](<(https://github.com/adobe/spectrum-web-components/blob/main/CONTRIBUTING.md)>)** document.
-   [x] I have added tests to cover my changes.
-   [x] All new and existing tests passed.
-   [x] I have reviewed at the Accessibility Practices for this feature, see: [Aria Practices](https://www.w3.org/TR/wai-aria-practices/)

## Best practices

This repository uses conventional commit syntax for each commit message; note that the GitHub UI does not use this by default so be cautious when accepting suggested changes. Avoid the "Update branch" button on the pull request and opt instead for rebasing your branch against `main`.

### CODE REVIEW

File: packages/icon/src/Icon.ts
Comment (Westbrook) [inline packages/icon/src/Icon.ts:78]: The default values for the event options in the event that you are redispatching here are all `false`. Can you share a little what you expect to do with these value as `true`?

Specifically, you've not added any code to support an application interacting with the `cancelability` of the event, but I'm not sure even what you could do there is you decided to. Taking it out might be the right call.

For `bubbles` and `composed`, is being able to hear this message outside of the shadow root where the `<sp-icon>` element is that useful to a consuming developer? Answers here can help form the basis for a little documentation around this addition so that this feature can be well leveraged by others.

Comment (vikalpgupta) [reply packages/icon/src/Icon.ts:78]: I made the silly mistake of copy-pasting this code and setting these back to default again, let me know if you have any more suggestions, Thanks

File: packages/icon/test/icon.test.ts
Comment (Westbrook) [inline packages/icon/test/icon.test.ts:89]: Does WebKit pass on this test for you locally? CI is showing `expected spy to have been called with exact arguments { isTrusted: [Getter] }`. If you thought there was something about the event construction you thought could match that up, that would be great. However, a generic `error.called` or `error.calledOnce` would also be enough here as there's no real information in the error itself.

Comment (vikalpgupta) [reply packages/icon/test/icon.test.ts:89]: Seems the spy is not getting called at all, looking for a way to check whether the same is supported via Webkit

Comment (Westbrook) [reply packages/icon/test/icon.test.ts:89]: You may also try changing the order of the bindings here. While it would be a bit unexpected, the error could be dispatched synchronously with the erroneous `src` application, which currently happens just before the event binding.

Comment (vikalpgupta) [reply packages/icon/test/icon.test.ts:89]: Found a bug logged for the same on WebKit's end https://github.com/microsoft/playwright/issues/22332

Comment (vikalpgupta) [reply packages/icon/test/icon.test.ts:89]: @Westbrook Any suggestion on this, what could be done next if this is only failing for playwright webkit type ones

Comment (Westbrook) [reply packages/icon/test/icon.test.ts:89]: Like I mentioned above, `error.called` or `error.calledOnce` or `error.callCount` are all fine alternatives to `calledWithExactly` In my opinion.

Comment (vikalpgupta) [reply packages/icon/test/icon.test.ts:89]: But for the playwright webkit, the above suggestions are also failing, I hope we are good without this feature for the said type browser for now

Comment (Westbrook) [reply packages/icon/test/icon.test.ts]: I see, I didn't quite catch that. You can leverage the helpers available here: https://github.com/adobe/spectrum-web-components/blob/main/tools/shared/src/platform.ts#L52 to query the browser and `skip` the test just for WebKit. Please be sure to add a comment as to why and link back to that issue.

File: packages/icon/stories/icon.stories.ts
Comment (Westbrook) [inline packages/icon/stories/icon.stories.ts:80]: Please include the following to exclude this story from our VRT suite.
```
imageIcon.swc_vrt = {
    skip: true,
};
```

**Review by Westbrook — CHANGES_REQUESTED** [review]
This is looking good! Thanks for the contribution.

If you can get that one change to the story and then rebase the branch to `main`, the only thing left will be signing our CLA and this will be good to ship.

**Comment by vikalpgupta** [comment]
Closing the pull request to reflect the CLA signed check

**Comment by vikalpgupta** [comment]
Closing the pull request to reflect the CLA signed check

File: packages/icon/test/icon.test.ts
Comment (Westbrook) [inline packages/icon/test/icon.test.ts:92]: [This is why we wait for CI to run before merging](https://app.circleci.com/pipelines/github/adobe/spectrum-web-components/15630/workflows/576533f7-9d4d-428b-80a2-6c0c0ac71011/jobs/322293)... It looks like the timing on this test can be flaky (at least in Firefox). Hopefully, it's just timing an not something more complicated. Can you try this?
```suggestion
            await waitUntil(() => error.calledOnce, 'The error was thrown.');
```
This will give all of the browsers as much as two seconds for the error to be thrown. It _should_ be OK, as in some passes Firefox has passed, but it's hard to tell 100% what's going on from those logs.

**Review by Westbrook — CHANGES_REQUESTED** [review]
Looking good with the CLA signed. I've rebased to `main` and assuming everything runs fine we'll merge this in after that. Thanks for helping us work through the tests here so we can include this additional feature on the Icon pattern. 🙇🏼

File: packages/icon/test/icon.test.ts
Comment (Westbrook) [reply packages/icon/test/icon.test.ts]: You need to import this method.

**Review by Westbrook — APPROVED** [review]
I think we're good to go!! Thanks for the contribution.

**Pull request merged**
